File: ios/Classes/model/VideoCodecInfo.swift

/// Video codecs.
enum VideoCodec: Int {
  /// VP8 video codec.
  case VP8

  /// VP9 video codec.
  case VP9

  /// H264 video codec.
  case H264

  /// AV1 video codec.
  case AV1

  /// H265 video codec.
  case H265
}

/// Represents an information about video codec.
class VideoCodecInfo {
  /// Identifier of the HW accelerated.
  private var isHardwareAccelerated: Bool

  /// Codec kind of the video codec.
  private var codec: VideoCodec

  /// Initializes a new `VideoCodecInfo` with the provided data.
  init(isHardwareAccelerated: Bool, codec: VideoCodec) {
    self.isHardwareAccelerated = isHardwareAccelerated
    self.codec = codec
  }

  /// Converts this controller into a Flutter method call response.
  func asFlutterResult() -> [String: Any] {
    [
      "isHardwareAccelerated": self.isHardwareAccelerated,
      "codec": self.codec.rawValue,
    ]
  }
}

File: android/src/main/kotlin/com/instrumentisto/medea_flutter_webrtc/model/VideoCodecInfo.kt
package com.instrumentisto.medea_flutter_webrtc.model

/**
 * Video codec kind.
 *
 * @property value [String] representation of this enum which will be expected on the Flutter side.
 */
enum class VideoCodec {
  VP8,
  VP9,
  H264,
  AV1,
  H265;

  companion object {
    fun valueOfOrNull(name: String): VideoCodec? {
      return values().firstOrNull { it.name == name }
    }
  }
}

/**
 * Represents an information about video codec.
 *
 * @property isHardwareAccelerated Identifier of the HW accelerated.
 * @property codec video codec kind.
 */
data class VideoCodecInfo(
    val codec: VideoCodec,
    val isHardwareAccelerated: Boolean,
) {
  /** Converts this [VideoCodecInfo] into a [Map] which can be returned to the Flutter side. */
  fun asFlutterResult(): Map<String, Any> =
      mapOf("isHardwareAccelerated" to isHardwareAccelerated, "codec" to codec.ordinal)
}
